### PR TITLE
commonlib follow-up

### PR DESCRIPTION
**Pull request by kingthorin**
Pursuant to feedback from other core team members.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -40,12 +40,21 @@ zapAddOn {
...
     }
 }
 
 dependencies {
     implementation("org.json:json:20190722")
     // https://mvnrepository.com/artifact/com.nimbusds/nimbus-jose-jwt
     implementation("com.nimbusds:nimbus-jose-jwt:8.3")
-    implementation("org.zaproxy.addon:commonlib:1.0.0")
+
+    compileOnly("org.zaproxy.addon:commonlib:1.0.0")
```
Comment (preetkaran20) [inline build.gradle.kts:57]: Does ZAP depends on this addon ? if not then How is this library present in ZAP

File: build.gradle.kts [hunk 2]
```diff
@@ -40,12 +40,21 @@ zapAddOn {
 
     manifest {
         author.set("KSASAN preetkaran20@gmail.com")
+
+        dependencies {
+            addOns {
+                register("commonlib")
```
Comment (preetkaran20) [inline build.gradle.kts:46]: I didn't got this. Can you please help me understand the usage of this ?

File: build.gradle.kts (hunk 1 above)
Comment (kingthorin) [reply build.gradle.kts:57]: When JWT is installed it will be installed from marketplace as a dependency.

File: build.gradle.kts (hunk 2 above)
Comment (kingthorin) [reply build.gradle.kts:46]: It establishes that the JWT add-on depends on the commonlib addon.

There are a number of pscan packages doing this now as well.

File: build.gradle.kts (hunk 1 above)
Comment (preetkaran20) [reply build.gradle.kts:57]: ok, Does "load addon from file" will also do the same operation ?

File: build.gradle.kts (hunk 2 above)
Comment (preetkaran20) [reply build.gradle.kts:46]: ok. thanks for explaining, will read more on this.

File: build.gradle.kts (hunk 1 above)
Comment (kingthorin) [reply build.gradle.kts:57]: Yup yup, already tested 🙂

I guess it’d fail if you are on a disconnected system and somehow have the addon file.

**Comment by kingthorin** [comment]
Yup

**Pull request merged**